core/src/test/java/org/elasticsearch/common/xcontent/ObjectParserTests.java

```diff
@@ -593,6 +596,131 @@ public void testIgnoreUnknownArrays() throws IOException {
         assertEquals(s.test, "foo");
     }
 
+    /**
+     * test parsing fields with a random name
```

Comment:
I think it'd be more right to say that this tests support for parsing unknown fields.